libhb/decavcodec.c

```diff
@@ -1927,15 +1975,32 @@ static int decavcodecvInit( hb_work_object_t * w, hb_job_t * job )
         pv->threads = HB_FFMPEG_THREADS_AUTO;
     }
 
-#if HB_PROJECT_FEATURE_QSV
-    if (pv->qsv.decode)
+    if(w->hw_device_ctx)
     {
-        pv->codec = avcodec_find_decoder_by_name(pv->qsv.codec_name);
+        const char* name = hb_vce_decode_get_codec_name(w->codec_param);
```

Comment:
Would't be better to use the hwaccel framework like nvdec and videotoolboxdec? Don't try to copy the QSV implementation, because it was added before hwaccel, and it's doing custom things that hopefully will go away soon.

Reply:
The issue with standalone hw decoders is that they are missing features, and are never in sync with the built-in parsers and hwaccel decoders. For example HDR dynamic metadata are ignored, and others metadata too. It's a pain because you never know what works and what doesn't.

Reply:
That ultimately leads to a bad user experience when features that should be transparent and just work, don't.